js_task/script.js

```diff
@@ -0,0 +1,470 @@
+                    }
+                }
+                else if (param === 'dateFrom'){
+                    returningPosts = returningPosts.filter(post => post.createdAt >= filterConfig.dateFrom);
+                }
+                else if (param === 'dateTo'){
+                    returningPosts = returningPosts.filter(post => post.createdAt < filterConfig.dateTo);
+                }
+                else if (param === 'author'){
+                    returningPosts = returningPosts.filter(post => post.author === filterConfig.author);
+                }
+            }
+
+            returningPosts.sort(function (first, second) {
+                if (first.createdAt < second.createdAt){
+                    return 1;
+                }
+                else if (first.createdAt > second.createdAt){
+                    return -1;
+                }
+                else{
+                    return 0;
+                }
+            });
+
+            return returningPosts.slice(skip, skip + top);
+        }
+        else {
+            let returningPosts = posts.slice(skip, skip + top);
+
+            returningPosts.sort(function (first, second) {
+                if (first.createdAt < second.createdAt){
+                    return 1;
+                }
+                else if (first.createdAt > second.createdAt){
+                    return -1;
+                }
+                else{
+                    return 0;
+                }
+            });
+
+            return returningPosts;
+        }
+    }
+
+    function getPost(id) {
+        if (typeof id === 'string'){
+            return posts.find(post => post.id === id);
+        }
+        else {
+            console.log('Incorrect type of id. Use \'string\'');
+        }
+    }
+
+    function validatePost(post, params = ['id', 'description', 'author', 'createdAt', 'photoLink', 'hashTags', 'likes']) {
+        return params.every(function (param) {
+            switch (param) {
+                case 'id':
+                    if (!post.id || typeof post.id !== 'string'){
+                        return false;
+                    }
+                    break;
+                case 'description':
+                    if (!post.description || typeof post.description !== 'string' || post.description.length > 250){
+                        return false;
+                    }
+                    break;
+                case 'author':
+                    if (!post.author || typeof post.author !== 'string'){
+                        return false;
+                    }
+                    break;
+                case 'createdAt':
+                    if (!post.createdAt || Object.prototype.toString.call(post.createdAt) !== '[object Date]'){
+                        return false;
+                    }
+                    break;
+                case 'photoLink':
+                    if (post.photoLink){
```

Comment:
You can simplify it to `if (post.photoLink && typeof post.photoLink !== 'string') {`